lang/python/CMakeLists.txt

```diff
@@ -23,17 +23,30 @@ include(${SWIG_USE_FILE})
 
 set(python_libs)
 set(python_version)
+
+set(required_version)
+if(PYTHON3_REQUIRED_VERSION)
+    if(NOT PYTHON3_REQUIRED_VERSION MATCHES "^([0-9]+)(\\.[0-9]+(\\.[0-9]+)?)?$")
+        message(FATAL_ERROR "Invalid value for PYTHON3_REQUIRED_VERSION: Requires a valid version string \
+            Provide a version number following the format: major[.minor[.patch]]")
+    endif()
+    if ("${PYTHON3_REQUIRED_VERSION}" VERSION_LESS 3)
+        message(FATAL_ERROR "Invalid value for PYTHON3_REQUIRED_VERSION: Requires a Python version >= 3")
+    endif()
+    set(required_version ${PYTHON3_REQUIRED_VERSION} EXACT)
+endif()
+
 if("${CMAKE_VERSION}" VERSION_LESS "3.12.0")
     # This method of finding python libs has been deprecated since version 3.12.
     # If we are running with a greater CMake version, opt to use the Python3 package.
     set(Python_ADDITIONAL_VERSIONS 3.9 3.8 3.7 3.6 3.5)
```

Comment:
Python 3.10 will be released in Oct. Are you aware of any way to automate the addition of future versions (I did some quick research with no luck). The `find_package` "version" field has the support of specifying a range of versions starting from CMake 3.19

Reply:
Unfortunately there isn't any way to automate this other than updating the `Python_ADDITIONAL_VERSIONS`. We could write our own custom CMake package module script to be more automated, though that would require significant work.
I don't believe there is much value writing a custom find package module as what you're describing is essentially what the Python3 module does in versions of CMake  >= 12 (in our below case find_package(Python3)`). This module doesn't need us to statically list out python versions and is able to detect Python 3.10 when its present on the users system. 

Given that we support CMake >= 10, we unfortunately still need this case.

Reply:
> what the Python3 module does in versions of CMake >= 12 (in our below case find_package(Python3)`). This module doesn't need us to statically list out python versions and will be able to detect Python 3.10 when its present on the users system.

It's good to know the newer versions of cmake has that covered. I agree there's value to write a customized module only to support this case.